pkg/authentication/basic/htpasswd.go

```diff
@@ -30,59 +33,114 @@ type sha1Pass string
 // NewHTPasswdValidator constructs an httpasswd based validator from the file
 // at the path given.
 func NewHTPasswdValidator(path string) (Validator, error) {
+	h := &htpasswdMap{users: make(map[string]interface{})}
+
+	err := h.loadHTPasswdFile(path)
+	if err != nil {
```

Comment:
It's idiomatic to inline this
```suggestion
	if err := h.loadHTPasswdFile(path); err != nil {
```